### PR TITLE
fix(sec): avoid to display passwords in monitoring command line

### PR DESCRIPTION
**Pull request by kduret**
## Description

* Hide passwords of commad line in host details panel (resources status)
* Display command line and hide password when there is a space in a macro value

**Fixes** MON-6204 MON-6589

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 19.10.x
- [x] 20.04.x
- [x] 20.10.x
- [x] 21.04.x (master)

<h2> How this pull request can be tested ? </h2>

Add password macros and try to display command line in resources status page

### CODE REVIEW

File: src/Centreon/Domain/HostConfiguration/Interfaces/HostConfigurationReadRepositoryInterface.php [hunk 1]
```diff
@@ -78,6 +78,18 @@ public function hasHostWithSameName(string $hostName): bool;
      */
     public function getNumberOfHosts(): int;
 
+    /**
+     * Find the command of a host.
+     *
+     * A recursive search will be performed in the inherited templates in the
+     * case where the host does not have a command.
```
Comment (sc979) [inline src/Centreon/Domain/HostConfiguration/Interfaces/HostConfigurationReadRepositoryInterface.php:85]:
```suggestion
     * Recursively search in the inherited templates if no result found.
```

File: src/Centreon/Domain/HostConfiguration/Interfaces/HostConfigurationServiceInterface.php [hunk 2]
```diff
@@ -65,6 +65,15 @@ public function getNumberOfHosts(): int;
      */
     public function findHostTemplatesRecursively(Host $host): array;
 
+    /**
+     * Find the command of a host.
```
Comment (sc979) [inline src/Centreon/Domain/HostConfiguration/Interfaces/HostConfigurationServiceInterface.php:69]:
```suggestion
     * Find the command of a host.
     * A recursive search will be performed in the inherited templates in the
     * case where the host does not have a command.
```

**Review by sc979 — DISMISSED** [review]

**Review by sc979 — DISMISSED** [review]

**Review by jeremyjaouen — DISMISSED** [review]

**Review by sc979 — DISMISSED** [review]

File: src/Centreon/Domain/Monitoring/CommandLineTrait.php [hunk 3]
```diff
@@ -24,73 +24,124 @@
...
+
+        return $configurationCommand;
+    }
+
+    /**
+     * Extra macro names from configuration command line
+     * example : ['$HOSTADDRESS$', '$_SERVICEPASSWORD$']
+     *
+     * @param string $commandLine
+     * @return array The list of macro names
```
Comment (callapa) [inline src/Centreon/Domain/Monitoring/CommandLineTrait.php:91]:
```suggestion
     * @return string[] The list of macro names
```

File: src/Centreon/Domain/Monitoring/MonitoringService.php [hunk 4]
```diff
@@ -336,117 +338,94 @@ public function findCommandLineOfService(int $hostId, int $serviceId): ?string
...
-                    }
-                }
-                $monitoringService->setCommandLine(implode(' ', $monitoringToken));
-            } else {
-                throw new MonitoringServiceException(
-                    'Different number of tokens between configuration and monitoring command line'
-                );
-            }
+        /**
+         * @var HostMacro[]|ServiceMacro[] $macros
```
Comment (callapa) [inline src/Centreon/Domain/Monitoring/MonitoringService.php:416]: You could create an interface that combines these two classes.

**Review by callapa — DISMISSED** [review]

**Review by callapa — DISMISSED** [review]

**Review by callapa — APPROVED** [review]

**Review by jeremyjaouen — APPROVED** [review]

**Pull request merged**